### ISSUE TITLE
Reading `__v` from the outside of the class doesn't give any error while Python interpreter gives error

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20502]
*Memo:
- mypy --strict test.py
- mypy 1.19.1
- python test.py
- Python 3.14.0
- Windows 11

Reading `__v` from the outside of the class doesn't give any error as shown below:

*Memo:
- A double leading underscore(__abc) can make attributes private in a class.
- Python interpreter gives error properly.

```python
class Cls:
    __v = 100

cls: Cls = Cls()

print(cls.__v) # No error
```

So, it should give an error like Python interpreter gives as shown below:

> error: 'Cls' object has no attribute '__v'

**Comment by A5rocks** [comment]
Duplicate of https://github.com/python/mypy/issues/8267
